scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py

```diff
@@ -182,3 +182,9 @@ def handle_rebin_options_set(self):
             self._view.set_raw_checkbox_state(False)
         else:
             self._view.set_raw_checkbox_state(True)
+
+    def handle_instrument_changed(self):
+        if self._model.plot_figure is not None:
+            from matplotlib import pyplot as plt
+            plt.close(self._model.plot_figure)
+        self._model.close_plot()
```

Comment:
I don't think this call is needed as close_plot is automatically linked up to the close signal on the figure window.

Reply:
it does not work without these lines